### PR TITLE
Update to Android Embedding V2

### PR DESCRIPTION
**Pull request by Lyokone**
Update the plugin to support android embedding v2.

### CODE REVIEW

**Comment by Lyokone** [comment]
Hello @mehmetf, If you have some time to review the changes, I'll be glad to hear from you before merging this :)

**Comment by mehmetf** [comment]
This is a fairly big PR so it is difficult for me to review. Did you do a lot of refactoring as well when adding v2 support?

**Comment by Lyokone** [comment]
Yeah It's kind of a big PR, no problem if you cannot review it.
I did separate the Android code in 4 parts so it's easier to support.

File: android/src/main/java/com/lyokone/location/LocationPlugin.java [hunk 1]
```diff
@@ -1,491 +1,201 @@
...
+        }
 
-    // Parameters of the request
-    private static long update_interval_in_milliseconds = 5000;
-    private static long fastest_update_interval_in_milliseconds = update_interval_in_milliseconds / 2;
-    private static Integer location_accuray = LocationRequest.PRIORITY_HIGH_ACCURACY;
-    private static float distanceFilter = 0f;
+        @Override
+        public void onStop(@NonNull LifecycleOwner owner) {
+            onActivityStopped(thisActivity);
```
Comment (mehmetf) [inline android/src/main/java/com/lyokone/location/LocationPlugin.java]: If you are `ActivityAware` why do you still need this? Can you stop these services `onDetachedFromActivity`?

It also doesn't look like `onActivityDestroyed` and `onActivityStopped` functions are doing anything useful.

Comment (Lyokone) [reply android/src/main/java/com/lyokone/location/LocationPlugin.java]: Thanks, it is indeed not needed. I wasn't able to get the application to not crash on exit, but It was resolved without this :)

**Pull request merged**